### ISSUE TITLE
Example of Supervised Regression

### ISSUE THREAD

**Issue by sadsadcool** [blei-lab/edward#647]
I'm a beginner of Edward. When I tried the example of supervised_regression.ipynb, I think the code in In[8] is wrong.

I think `  w_samples = w.sample(n_samples)[0].eval()` should be `  w_samples = w.sample(n_samples)[:, 0].eval()`

I found it when I changed n_samples. The first feature dimension is given in the first column rather than first row.

**Comment by closedLoop** [comment]
I also agree with this.  I have also created a visualizations of the betas to better confirm that they have converged and they use the same slicing as mentioned above

```python
def visualize_dists(w, qw, actual=None, n_samples=1000, figsize=None):
    """
    Creates a grid of histograms of samples from two distributions 'w' and 'qw'
    Optionally creates a red vertical line for each value in 'acutal'
        
    Args:
        w:  the prior variable with an ability to sample  
        qw:  the posterior variable with the same dimensions as w
        actual:  a list-like of known true values, has the same dimension as w
        n_samples (int): the number of samples, default = 1000 
        figsize (tuple): a tuple of the figure size, default is (20, 2*D) where D is the dimension of w 

    Returns:
        matplotlib.figure: A gr
        
    Usage:
        visualize_dists(w, qw, actual=[0.1, 0.0, 0.23, 0.5, 0.3], n_samples=1000)
    """
    import seaborn as sns
    import matplotlib.pylab as plt
    
    w_samples = w.sample(n_samples).eval()
    qw_samples = qw.sample(n_samples).eval()

    D = w_samples.shape[1]

    if figsize is None:
        figsize = (20, D * 2)

    fig = plt.figure(figsize=figsize)
    axes = []
    for i in range(D):
        if i == 0:
            ax = fig.add_subplot(np.ceil(D / 2), 2, i + 1)
        else:
            ax = fig.add_subplot(np.ceil(D / 2), 2, i + 1)
        sns.distplot(w_samples[:, i], kde=False, rug=False, ax=ax, color='grey')
        sns.distplot(qw_samples[:, i], kde=False, rug=False, ax=ax, color='b')
        ax.set_title('variable {}'.format(i))
        axes.append(ax)

    if not actual is None:
        for i in range(D):
            ymin, ymax = ax.get_ylim()
            axes[i].vlines([actual[i]], ymin=ymin, ymax=ymax, color='r')

    fig.tight_layout()
    fig.suptitle('prior vs posterier weight samples')
    return fig

visualize_dists(w, qw, actual=w_true, n_samples=1000)
```
![untitled](https://cloud.githubusercontent.com/assets/2898137/26469494/2b142f4a-4168-11e7-9874-c3fb64c9de34.png)

**Comment by dustinvtran** [comment]
Fixed in #649.
